### PR TITLE
BasePage refactored. Login page was added.

### PR DESCRIPTION
**Pull request by diogoti2005**
Signed-off-by: adiogoti <andrei.diogoti@endava.com>

### CODE REVIEW

File: src/main/java/com/demoqa/shop/pages/AccountPage.java [hunk 1]
```diff
@@ -18,18 +20,24 @@
     public WebElement button_register;
     @FindBy(how = How.XPATH, using = "//p[@class='login message']")
     public WebElement message_login;
+    @FindBy(how = How.XPATH, using = "//input[@id='username']")
+    public WebElement field_username_login;
+    @FindBy(how = How.XPATH, using = "//input[@id='password']")
+    public WebElement field_password_login;
```
Comment (en-nlasco) [inline src/main/java/com/demoqa/shop/pages/AccountPage.java]: In my opinion all webelements from the page should be private, and have each corresponding public method, that will access correctly this webelement(buttons- click, textfields-sendkeys, and not vice versa by error)

File: src/main/java/com/demoqa/shop/pages/HomePage.java [hunk 2]
```diff
@@ -7,24 +7,36 @@
...
+    @FindBy(how = How.XPATH, using = "//a[contains(text(),'My Wishlist')]")
+    public WebElement lnk_MyWishlist;
+    @FindBy(how = How.XPATH, using = "//a[contains(text(),'Checkout')]")
+    public WebElement lnk_Checkout;
+    @FindBy(how = How.XPATH, using = "//span[contains(@class,'cart-name-and-total')]")
+    public WebElement lnk_Cart;
+    @FindBy(how = How.XPATH, using = "//a[contains(@class,'noo-search')]")
+    public WebElement lnk_Search;
 
     public HomePage(WebDriver driver) {
```
Comment (en-nlasco) [inline src/main/java/com/demoqa/shop/pages/HomePage.java:48]: here we have public method, maybe would be better to keep lnk_Dismiss private

File: src/main/java/com/demoqa/shop/pages/AccountPage.java (hunk 1 above)
Comment (diogoti2005) [reply src/main/java/com/demoqa/shop/pages/AccountPage.java]: all webelements from the page are set as private and corresponding public method

File: src/main/java/com/demoqa/shop/pages/HomePage.java (hunk 2 above)
Comment (diogoti2005) [reply src/main/java/com/demoqa/shop/pages/HomePage.java:48]: lnk_Dismiss is  private now

**Review by en-nlasco — APPROVED** [review]

**Review by nluchian — APPROVED** [review]

**Pull request merged**